src/main/kotlin/dev/polek/adbwifi/services/PinDeviceService.kt

```diff
@@ -19,6 +19,12 @@ class PinDeviceService : PersistentStateComponent<PinDeviceService> {
     @OptionTag(converter = PinnedDeviceListConverter::class)
     var pinnedDevices: List<PinnedDevice> = listOf()
 
+    fun addPreviouslyConnectedDevice(device: Device) {
+        if (device.connectionType == WIFI && device.address?.ip!!.isNotEmpty() && !pinnedDevices.contains(device)) {
+            pinnedDevices.add(device)
+        }
+    }
```

Comment:
Because of `device.connectionType == WIFI` condition, this method does not add the device to the list if it's a `USB` or `Previously connected` device.
In the case of "Previously connected device", it gets removed from the list because  `RenameDeviceDialogWrapper#renameDevice()` also calls `pinService.removePreviouslyConnectedDevice(...)`.

Reply:
Updated in: [2c956c3](https://github.com/y-polek/ADB-Wi-Fi/pull/21/commits/2c956c3828f2a15aee6c89a0f24a2b8cbee7ab0e)